scripts/verify-artifact-contents

```diff
@@ -35,19 +35,19 @@ done
 # (https://kotlinlang.org/docs/whatsnew14.html#dependency-on-the-standard-library-added-by-default),
 # but kotlin-reflect needs to be added by us explicitly since it's required in the runtime.
 expected_libraries=(org.eclipse.jgit reflections slf4j-lambda-core vavr vavr-match)
-for lib in ${expected_libraries[@]}; do
+for lib in "${expected_libraries[@]}"; do
   grep -qEx "${lib//./\\.}-[0-9.]+.*\.jar" <<< "$jars_in_plugin_zip" || fail "$lib jar is expected but missing from $plugin_zip"
 done
 
 # We don't want either slf4j-api or any SLF4J implementation jars in the final plugin artifact
 # as this might lead to a linker error (for api) or an SLF4J error (for impl) in the runtime.
 forbidden_library_prefixes=(checker ide-probe idea junit-driver kotlin lombok remote-robot scala slf4j)
-for lib in ${forbidden_library_prefixes[@]}; do
+for lib in "${forbidden_library_prefixes[@]}"; do
   if matching_jars=$(grep "^$lib" <<< "$jars_in_plugin_zip"); then
     for jar in $matching_jars; do
       jar_base_name=${jar/-[0-9]*/}
       # Maybe the library matches the forbidden prefix but is still expected (e.g. slf4j-lambda-core)?
-      if ! [[ " ${expected_libraries[@]} " == *\ $jar_base_name\ * ]]; then
+      if ! [[ " ${expected_libraries[*]} " == *\ $jar_base_name\ * ]]; then
```

Comment:
What's the difference?

Reply:
https://github.com/koalaman/shellcheck/wiki/SC2199. Here it doesn't matter, we do want an implicit concatenation anyway.

In general:
`$@`, `${my_array[@]}`: each element is a separate string
`$*`, `${my_array[*]}`: all elements concatenated into a single string

Showcase:

```shell

array=(foo bar qux)

function print_args() {
  echo "$1:"
  shift
  for arg in "$@"; do
    echo " - $arg"
  done
  echo
}

print_args '${array[@]}' "${array[@]}"
print_args '${array[*]}' "${array[*]}"

function call_print_args() {
  print_args '$@' "$@"
  print_args '$*' "$*"
}

call_print_args foo bar qux
```

Output:
```
${array[@]}:
 - foo
 - bar
 - qux

${array[*]}:
 - foo bar qux

$@:
 - foo
 - bar
 - qux

$*:
 - foo bar qux
```